docs/spec_draft.md

```diff
@@ -550,6 +551,64 @@ IEEE-754 specification.
 
 [Back to Ops](#index-of-documented-ops)
 
+## stablehlo.iota
+
+### Semantics
+Fills a `result` array with values in increasing order starting from zero along
+the `iota_dimension` dimension.
+
+Note: For integers, if the dimension size is larger than what the element type's
+maximum value can hold, an overflow occurs and the behavior is implementation-
+defined and one of the following:
+  * mathematical result modulo $2^n$, where n is the bit width of the result,
+  for unsigned overflow/underflow. For signed integer overflow/underflow, wraps
+  the result around the representable range $[-2^{n-1},\ 2^{n-1} - 1]$.
+  * saturation to $2^{n-1} - 1$ (or $-2^{n-1}$) for signed overflow (or signed
+  underflow) and saturation to $2^n - 1$ (or $0$) for unsigned overflow (or
+  unsigned underflow).
+
+### Operands
+
+| Name             | Type   |
+|------------------|--------|
+| `iota_dimension` | `ui64` |
+
+### Results
+
+| Name     | Type                                                       |
+|----------|------------------------------------------------------------|
+| `result` | tensor of integer, floating-point or complex element types |
+
+### Constraints
+
+  * (C1) `iota_dimension` >= 0
+  * (C2) `result` rank >= 1
+  * (C3) `result` rank > `iota_dimension`
+
+### Examples
+
+```mlir
+%result = "stablehlo.iota"() { iota_dimension = 0 } : (tensor<i64>) -> tensor<4x5xi32>
```

Comment:
`%result = "stablehlo.iota"() { iota_dimension = 0 } : () -> tensor<4x5xi32>` is the correct syntax. In the future PRs, we should run `stablehlo-opt` on all code examples - I'll add that to the checklist.